Graph Explorer docs update (#17622)
microsoftgraph/microsoft-graph-docs: concepts/graph-explorer/graph-explorer-features.md

---
title: Work with Graph Explorer
description: Use Graph Explorer to make Microsoft Graph REST API requests and view corresponding responses. Learn how to use some of the important features in Graph Explorer.
ms.localizationpriority: high
author: RabebOthmani
---

# Work with Graph Explorer

> [Graph Explorer](https://developer.microsoft.com/graph/graph-explorer/) is a developer tool that lets you learn about Microsoft Graph APIs. It has a variety of features to help you on your journey. This article describes Graph Explorer's main interface areas and how to use some of the important features.

Reviewer: I suggest that we revise the last sentence as follows:

This article describes some of the important features of Graph Explorer and how to use them.

For an overview, see [Use Graph Explorer to try Microsoft Graph APIs](/graph/graph-explorer/graph-explorer-overview).

## Header

> The header allows you to sign in, view the tenant you are using and access your settings, helpful links and the survey to provide feedback for the tool.

Reviewer: If you keep this section, revise the text: "From the top navigation, you can sign in, view the tenant you're using, access your settings, and find helpful links and a feedback mechanism." The current sentence structure anthropomorphizes "header". Also we should refer to it as "top navigation". A header is like an H1, H2, etc. in an article.

![Screenshot of Graph Explorer header](./images/header-screenshot.png)

- **The avatar** : sign in, view your profile, consent to permissions, switch accounts or sign out of your account. For more on consenting to permissions, see [Consent to permissions](#consent-to-permissions).
- **Feedback** : If you wish to provide feedback, you can fill in the survey form.
- **Help** : Access useful links such as the documentation or the GitHub repo for Graph Explorer.
- **Settings** : Access Graph Explorer settings to change the interface theme or to get a free instant sandbox preconfigured with sample data packs
- **Tenant** : View the tenant you are currently using.

> ### Consent to permissions

Reviewer: The structure of the article is currently a bit confusing, as we seem to have H2 headings for noun phrases that represent some element of the UI, and task-based H3 subsections. We need to think about a structure that makes the intent of the article clear to the reader. Either its a description of the UI elements and what they do, or it's a how-to article about using the tool to perform tasks. I recommend keeping it as the latter, which is the structure of the currently published topic and reflects the title "Work with Graph Explorer". Will readers use GE section by section, like go to the Header, then the Sidebar, etc., or will they want to complete a task and follow the UI to do so?

> The user or administrator must grant Graph Explorer the correct permissions by using a consent process to access data in Microsoft Graph. You can consent to permissions in Graph Explorer by choosing either the **Modify permissions** tab or the **Consent to permissions** option under the profile avatar when you’re signed in. The **Modify permissions** tab lists all the permissions that you need to run the query in the address bar.

Reviewer 2: Line 28: The last sentence can be misleading. A reader could interpret that they need all the permissions listed, and request all of them. Could we clarify that the tab lists the permissions from least to most privileged and remind users to use the least privileged when possible.

Author: Done

> The user or administrator must grant Graph Explorer the correct permissions by using a consent process to access data in Microsoft Graph. You can consent to permissions in Graph Explorer by choosing either the **Modify permissions** tab or the **Consent to permissions** option under the profile avatar when you’re signed in. The **Modify permissions** tab lists all the permissions that might let you run the query in the address bar. When possible, choose the least privileged permission.

Reviewer: The use of "might" here adds some ambiguity; can we change it to "will"?

3. See the list of permissions required to run the query.
4. Select the **Consent** button next to the permission that you want to consent to.

> ![Screenshot of Graph Explorer with the steps to consent to permissions highlighted](./images/modify-permissions-screenshot.png)

Reviewer 2: Please add alt text to all images in this topic. Some of the images precede a paragraph that provides a good explanation so, in those cases a simple alt text would be enough.

Author: @isvargasmsft I'm not clear on this. All images are presented as ![ALT TEXT](image) , isn't that enough

The **Modify  permissions** feature is currently in preview, and some queries might be missing permissions. If permissions are missing for a query, the **Consent to permissions** option in the avatar gear contains the list of all available permissions:

> 1. Go the avatar gear and choose the **Consent to permissions** option. This option contains the list of all available permissions.

Reviewer: What is the "avatar gear"?

2. From the list of all the permissions, consent to the ones you want.

![Screenshot of Graph Explorer with the Select permissions option highlighted](./images/consent-permissions.png)

> ## Sidebar

Reviewer: Is the the correct term? Seems like a more standard term for this part of the tool would be "Left pane".

The sidebar allows you to access the main features of Graph Explorer:

> - [Sample queries](#get-started-with-microsoft-graph-apis)

Reviewer: If you're going to have jump links within the article, the link text should match the heading that its linking too. Currently this is an unclear experience and makes the reader think that the link is going to the wrong destination - the relationship between the link text and the target isn't clear.

- [Resources](#explore-all-of-the-microsoft-graph-resources)

> - [History](#store-and-share-queries)

Reviewer 2: Sidebar section: Could you please elaborate a bit more and/or add a screenshot of the sidebar. For a user whose not familiar with the UI the section is not providing enough guidance to understand what the sidebar is, where to find it and what is it for.

Reviewer: Per Isaac's comment, I think it will be more clear to refer to it as the left pane.

### Get started with Microsoft Graph APIs

> To get started with Microsoft Graph and get yourself familiar with Graph Explorer, the **Sample queries** feature provides a set of queries categorized by service that you can select and run.

Reviewer: Revise as: "The **Sample queries** tab provides a set of queries categorized by service that you can select and run to help you get started with Microsoft Graph API calls."

> To get started with Microsoft Graph and get yourself familiar with using Graph Explorer, the **Sample queries** feature provides a set of queries categorized by service that you can select and run.

Reviewer 2: Suggestion for simplicity:
```suggestion
To get started with Microsoft Graph and get yourself familiar with Graph Explorer, the **Sample queries** feature provides a set of queries categorized by service that you can select and run.
```

![Screenshot of the sample queries tab](./images/sample-queries-screenshot.png)

> The query box will be populated by the selected query. You can then make the request by clicking the Run query button.

Reviewer: Revise this text as:
"When you select a query, the selected query populates the query box. To try the request, choose **Run query**.

> ### Explore all of the Microsoft Graph resources

Reviewer: Revise as "Explore resources".

> The Resources feature allows you to explore the resources on Microsoft Graph in their entirety, on both v1.0 and beta versions . Navigate the tree to view a specific resource or alternatively, you can search for it.

Reviewer 2: Suggestion for voice:
```suggestion
The Resources feature allows you to explore the resources on Microsoft Graph in their entirety, on both v1.0 and beta versions. Navigate the tree to browse for a specific resource or, alternatively, use the search bar to find it.
```

> The Resources feature allows you to explore the resources on Microsoft Graph in their entirety, on both v1.0 and beta versions. Navigate the tree to browse for a specific resource or, alternatively, use the search bar to find it.

Reviewer: Revise as: "Use the **Resources** tab to explore Microsoft Graph resources in the beta and v1.0 endpoints. Browse the list of resources or use the search bar to search for a specific resource."

![Screenshot of the resources tab](./images/resources-tab-screenshot.png)

> With the **Resources** tab you can export a collection of resources into a Postman collection:

Reviewer: Revise as: "From the **Resources** tab, you can export a collection of resources into a Postman collection by doing the following:".

> With Resources Explorer, you can export a collection of resources you are interested in, into a Postman collection:

Reviewer 2: Line 68: In general, we avoid capitalizing names that are not TMd brands. Adding a suggestion for simplicity also:
```suggestion
With the **Resources** tab you can export a collection of resources into a Postman collection:
```

1. From the kebab menu next to the resource, select **Add to collection**.

> 1. Once you complete adding resources, click on the **Preview collection** button.

Reviewer: Revise as: "When you're finished adding resources, choose **Preview collection**.

> 1. From the collection view, you can review the resources and remove any if you chose to.

Reviewer: Revise as: "From the collection view, you can review the resources you selected and remove any if you choose."

> 1. Click on **Download Postman collection** to download your collection.

Reviewer: Revise as: "Choose **Download Postman collection**."

> 1. Click on the **Download Postman collection** to download your collection.

Reviewer 2: Edit:
```suggestion
1. Click on **Download Postman collection** to download your collection.
```

![Screenshot of the preview collection screen](./images/download-collection-screenshot.png)

### Store and share queries

> All queries you run in Graph Explorer are saved for 30 days in the **History** tab. In the **History** tab, you can:

Reviewer: In the second sentence, change "In" to "From".

> Queries that are run in Graph Explorer are saved for 30 days on the **History** tab. On the **History** tab, you can:

Reviewer 2: Edit for simplicity:
```suggestion
All queries you run in Graph Explorer are saved for 30 days in the **History** tab. In the **History** tab, you can:
```

- Export all history items in **.har** format.
- Delete all history items.
- View a history item.
- Run a query.
- Export a history item in **.har** format.
- Delete a history item.

![Screenshot of the History tab](./images/history-tab-screenshot.png)

> To share queries that you run, select the **Share** button situated next to the Run query button, and then select **Copy**. This copies a link to share that allows others to see your query and the results.

Reviewer 2: Edit:
```suggestion
To share queries that you run, select the **Share** button situated next to the Run query button, and then select **Copy**. This copies a shareable link that allows others to see your query and the results.
```

![Screenshot of Graph Explorer with the Share and Copy options highlighted](./images/share-query-screenshot.png)

> ## Main work area

Reviewer: Can we call this something a little bit more clear, like "Query pain"?

> This area is where you will be spending most of your time on Graph Explorer. This area has everything you need to run a request:

Reviewer: Revise as: Use the query pane to run requests and see the responses. The query pane includes the following elements:".

We shouldn't imply that we know where our audience is going to be spending their time. ;)

> This area is where you will be spending most of your time on Graph Explorer. This area has everything you need to run a request:

Reviewer 2: Line 96: What if I spend most of my time creating resource collections for Postman? 😆 
Can we rephrase this intro sentence to make it more focused on the value of the section?

> - **Query box**: you can manually build your query.

Reviewer: The text after the colons in this bulleted list should consistently be capitalized. Also, please make the descriptions consistent, either noun or verb phrases. Here, the description seems incomplete - they can manually specify a query, but it also gets autopopulated if they select a query, right?

> - **HTTP verbs**: a dropdown list of the HTTP methods.

Reviewer: So for example, if you wanted to consistently use verb phrases, you can revise this to say "Choose the HTTP verb from the dropdown list." Also, hmm, the UI element isn't actually called "HTTP verbs", so you need a different way to describe this; otherwise it's going to mislead the reader.

> - **API version**: a dropdown list of the API versions available publicly.

Reviewer: This also isn't the the name of the UI element.

> - **Modify permissions**: you need to consent to the right permissions to be able to run a query. For details, see [consenting to permissions](#consent-to-permissions).

Reviewer: For completeness, should we also include Request body and Request headers?

> - **Modify permissions**: you need to consent to the right permissions to be able to run a query. Read more about [consenting to permissions](#consent-to-permissions).

Reviewer 2: Edit for voice:
```suggestion
- **Modify permissions**: you need to consent to the right permissions to be able to run a query. For details, see [consenting to permissions](#consent-to-permissions).
```

> - **Access token**: this tab shows your access token when you're signed in. You can copy the token if you need to use it in your favorite REST client application.

Reviewer 2: Lines 98 to 102: In general we try to apply the principles of parallel writing. That means, for a list of sentences related, we try to compose them using a similar format. For example, if we start with infinitive, all the other should too. If we start with a verb in 3rd person present, the others should too. In this case, I'd suggest starting in 3rd person present. For example (just an example to illustrate parallel writing, you can choose the way you prefer):
- Allows you to manually build a query. 
- Lists all the HTTP methods. 
- Indicates the API version for the request.

>  ![Screenshot of Graph Explorer highlighting the main work area](./images/run-query-tabs-screenshot.png)

Reviewer: Change "main work area" to "query pane" here too.

> This is also where you see:

Reviewer: This is kind of repetitive. Can we revise as something like: "When you run a query, you will see a **Response preview** and **Response headers** in the response pane." And then remove the two bullets.

- **Response preview**: You can see the response body here.

> - **Response headers**: You can see the response headers here.

Reviewer 2: Lines 106 to 109: Could you please elaborate a bit more? This section is not providing enough guidance and it is quite important. For example, you can start with an H3 Visualize the response, and explain a bit further what to expect from the response preview and response headers tabs. Talk about the format, how to read the response, and how to interact with it. A user may or may not know that the response preview comes in JSON format and doesn't include the headers, which go in the headers tab. And I also noticed you can expand and collapse sections of the response body with the small arrows at the left. 

All of these is material for the section :)

Reviewer 2: In addition, the other tabs are receiving their own H3s. If the intention of this topic is to give readers a tour by the product, it is a bit of a broken experience to give a different level of attention to tabs in the same category (visually speaking at least).

> On top of query related features, the main work area also contains:

Reviewer: Revise as: "In addition to the response, this pane also includes:".

> - [Microsoft Graph Toolkit Adaptive cards integration](#microsoft-graph-toolkit-integration)

Reviewer: I think we need to remove "Adaptive cards" from the link text here.

- [Adaptive cards integration](#adaptive-cards-integration)

> - [Code snippets](#leverage-the-power-of-the-client-libraries)

Reviewer: We should following the order of the UI from left to right, so put code snippets first, and also rearrange the corresponding H3 sections to use the same order.

### Microsoft Graph Toolkit integration

The [Microsoft Graph Toolkit](../toolkit/overview.md) is a collection of reusable, framework-agnostic web components and helpers for accessing and working with Microsoft Graph. The components are fully functional, with built-in providers that authenticate with and fetch data from Microsoft Graph.

> Graph Explorer allows you to visualize Microsoft Graph Toolkit components that correspond to the API queries. A blue dot on the **Toolkit component** tab indicates that the Toolkit provides a component for the currently specified REST API query in Graph Explorer. You can conveniently copy the code for the component to your app.

Reviewer: When we refer to the toolkit instead of Microsoft Graph Toolkit, toolkit is lowercase.

> Graph Explorer allows you to see Microsoft Graph Toolkit components that correspond to the API queries. A blue dot on the **Toolkit component** tab indicates that the Toolkit provides a component for the currently specified REST API query in Graph Explorer. You can conveniently copy the code for the component to your app.

Reviewer 2: Edits:
```suggestion
Graph Explorer allows you to visualize Microsoft Graph Toolkit components that correspond to the API queries. A blue dot on the **Toolkit component** tab indicates that the Toolkit provides a component for the currently specified REST API query in Graph Explorer. You can conveniently copy the code for the component to your app.
```

In addition, for accessibility purposes, could you mention that at the right you have the code for the component and a visualization of how the component renders at the left?

![Screenshot showing the Toolkit components tab](./images/toolkit-screenshot.png)

### Adaptive cards integration

> [Adaptive cards](https://adaptivecards.io/) are platform-agnostic snippets of UI, authored in JSON, that apps and services can openly exchange. When you run a query and an adaptive card is available, a blue dot appears on the **Adaptive cards** tab.

Reviewer 2: In general, we avoid attributing animated behavior to non-animated objects. In this case, saying that something "appears".
```suggestion
[Adaptive cards](https://adaptivecards.io/) are platform-agnostic snippets of UI, authored in JSON, that apps and services can openly exchange. A blue dot on the **Adaptive cards** tab indicates that an adaptive card is available for the selected query.
```
